snomed/com.b2international.snowowl.validation.snomed/src/main/resources/scripts/ruleSnomedCommon4.groovy

```diff
@@ -35,9 +41,33 @@ SnomedConceptSearchRequestBuilder conceptsRequestBuilder = SnomedRequests.prepar
 		.setExpand("preferredDescriptions()")
 				
 if (params.isUnpublishedOnly) {
-	conceptsRequestBuilder.filterByEffectiveTime(EffectiveTimes.UNSET_EFFECTIVE_TIME)
+
+	def unpublishedLangueMembers = SnomedRequests.prepareSearchMember()
+		.filterByReferencedComponentType(SnomedTerminologyComponentConstants.DESCRIPTION_NUMBER)
+		.filterByRefSetType(SnomedRefSetType.LANGUAGE)
+		.filterByEffectiveTime(EffectiveTimes.UNSET_EFFECTIVE_TIME)
+		.filterByActive(true)
+		.all()
+		.build()
+		.execute(ctx)
+		.collect({ it.referencedComponent.id})	
+	
+	def conceptsWithUnpublishedLanguageMembers = SnomedRequests.prepareSearchDescription()
+				.filterByIds(unpublishedLangueMembers)
+				.all()
```

Comment:
You can safely set the limit to `unpublishedLangueMembers.size()` instead of requesting `all()`.